### DISCUSSION TITLE
Is there a way we can reset all the atoms? With a single function?

### DISCUSSION THREAD

**Discussion by bhushan-patil-official** [pmndrs/jotai#1540] [Q&A]
On logout we want to reset all the states, currently what we are doing is manually resetting the states. Is there a function with which we can reset all the state in one go? Or select some states to reset with just one go?

**Comment by dai-shi** [comment]
Surprisingly, the library has no knowledge about "all" atoms. (This avoids memory leaks.)
You could re-create a store with re-mounting `<Provider>` with using `key={}`, but it might be tricky for logout action.

You can create a write-only atom to write to multiple atoms at once.
```js
const countAtom = atom(0)
const textAtom = atom('')

const resetAtom = atom(null, (get, set) => {
  set(countAtom, 0)
  set(textAtom, '')
})
```

**Reply by bhushan-patil-official** [reply]
Woah, thanks, I will try this tonight and let you know.
